tests/test_remote_temp_cleanup.py

```
@@ -0,0 +1,142 @@
+    mock_dag.depending[producer] = {consumer: {input_file}}
+    mock_dag._finished.add(consumer)
+    mock_dag._needrun = set()
+
+    mock_dag.workflow.storage_settings.notemp = False
+    mock_dag.workflow.dryrun = False
+
+    # Collect unneeded files via the same logic as handle_temp
+    from functools import partial
+    from itertools import filterfalse
+
+    is_temp = lambda f: f.flags.get("temp", False)
```

Ruff (0.15.12)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `is_temp` as a `def`

(E731)